src/utils/twoFactor.js

```diff
@@ -0,0 +1,181 @@
+
+export const getRequest = () => {
+    return JSON.parse(localStorage.getItem(`__multisigRequest`) || `{}`)
+}
+
+export const setRequest = (data) => {
+    localStorage.setItem(`__multisigRequest`, JSON.stringify(data))
+}
+
+export const sendTwoFactorRequest = async (wallet, accountId, method, requestId = -1, data = {}) =>  {
+    if (!accountId) accountId = wallet.accountId
+    if (!method) method = await wallet.get2faMethod()
+    // add request to local storage
+    setRequest({ accountId, requestId, data })
+
+    const serverResponse = wallet.postSignedJson('/2fa/send', {
+        accountId,
+        method,
+        requestId,
+        data
+    }).catch((e) => console.log('/2fa/send failed', e));
+    
+    if (requestId !== -1) {
+        return new Promise((resolve) => {
+            store.dispatch(promptTwoFactor((verified) => {
+                if (verified) {
+                    wallet.tempTwoFactorAccount = null
+                }
+                resolve(verified)
+            }))
+        })
+    } else {
+        return await serverResponse
+    }
+}
+
+export const twoFactorRequest = async (wallet, account, request) => {
+    if (wallet.tempTwoFactorAccount) account = wallet.tempTwoFactorAccount
+    const { accountId } = account
+    console.log('makeTwoFactorRequest', accountId, request, account)
+    const contract = getContract(account)
+    await deleteUnconfirmedRequests(contract)
+    const request_id = await getNextRequestId(contract)
+    await addRequestAndConfirm(contract, request)
+    const request_id_after = await getNextRequestId(contract)
+    if (request_id_after > request_id) {
+        const data = { request_id, request }
+        const method = await wallet.get2faMethod()
+        return await sendTwoFactorRequest(wallet, accountId, method, request_id, data)
+    }
+}
+
+export const twoFactorAddKey = (wallet, account, publicKey, contractId, fullAccess = false) => {
+    const request = {
+        receiver_id: account.accountId,
+        actions: [addKeyAction(account, publicKey, contractId, fullAccess)]
+    }
+    return twoFactorRequest(wallet, account, request)
+}
+
+export const twoFactorRemoveKey = (wallet, account, publicKey) => {
+    const request = {
+        receiver_id: account.accountId,
+        actions: [deleteKeyAction(publicKey)]
+    }
+    return twoFactorRequest(wallet, account, request)
+}
+/********************************
+Deploys 2/3 multisig contract with keys from contract-helper and localStorage
+@todo check account has enough near to actually do this
+********************************/
+export const twoFactorDeploy = async (wallet) => {
+    const accountData = await wallet.loadAccount()
+    // get multisig contract to deploy
+    const contractBytes = new Uint8Array(await fetch('/multisig.wasm').then((res) => res.arrayBuffer()))
+    // get account
+    const { accountId } = accountData
+    const account = wallet.getAccount(accountId)
+    // 1. get localstorage keys
```

Comment:
not sure if this comment matches what is happening.
Isn't `getAccessKeys()` getting access keys from server?

Reply:
clarified